packages/loopring_v3/contracts/amm/libamm/AmmJoinProcess.sol

```diff
@@ -188,8 +192,12 @@ library AmmJoinProcess
         // Calculate the amounts to deposit
         uint ratio = uint(AmmData.POOL_TOKEN_BASE).mul(mintAmount) / ctx.totalSupply;
 
+        uint newTotalSupply = ctx.totalSupply.add(mintAmount);
         for (uint i = 0; i < ctx.tokens.length; i++) {
             amounts[i] = (ratio.mul(ctx.tokenBalancesL2[i]) / AmmData.POOL_TOKEN_BASE).toUint96();
+
+            // Update virtual balances
+            ctx.vTokenBalancesL2[i] = (uint(ctx.vTokenBalancesL2[i]).mul(newTotalSupply) / ctx.totalSupply).toUint96();
```

Comment:
The virtual balances always scale according to the new LP token balance. I believe with this approach, joins and exists will change the 'current price' in the pool. Correct? 

If so, is it possible that someone joins the pool to change the price in favor for his next trade?

Reply:
No, joins/exits will keep the existing price. Like you said, the virtual balances are only scaled by joins/exits so the AMM price  will remain the same (because the AMM price if defined by the division of the virtual balances, which remain the same when  both virtual balances were multiplied by the same factor).

Reply:
You are right. I misunderstand it.